app/controllers/users_controller.rb

```diff
@@ -62,22 +62,11 @@ def registration_code
 
   def admin_code
     begin
-      flash[:notice] = User.change_admin_code(params[:admin_code])
+      flash[:notice] = User.change_code("admin", params[:admin_code])
     rescue ActiveRecord::RecordInvalid => e
       flash[:notice] = Code.code_uniqueness_message
     end
 
     redirect_to users_admin_path
   end
-
-  def login
```

Comment:
I was wondering about this. I suppose Devise handles them, so they're not needed.